mail.c

```diff
@@ -372,24 +398,15 @@ readmail(struct queue *queue, int nodot, int recp_from_header)
 		return (-1);
 
 	while (!feof(stdin)) {
-		if (fgets(line, sizeof(line) - 1, stdin) == NULL)
+		newline[0] = '\0';
+		if ((linelen = getline(&line, &linecap, stdin)) <= 0)
 			break;
-		if (had_last_line)
-			errlogx(EX_DATAERR, "bad mail input format:"
-				" from %s (uid %d) (envelope-from %s)",
-				username, useruid, queue->sender);
-		linelen = strlen(line);
-		if (linelen == 0 || line[linelen - 1] != '\n') {
-			/*
-			 * This line did not end with a newline character.
-			 * If we fix it, it better be the last line of
-			 * the file.
-			 */
-			line[linelen] = '\n';
-			line[linelen + 1] = 0;
-			had_last_line = 1;
-		}
 		if (!had_headers) {
+			if (linelen > 1000)
```

Comment:
Yes I followed the discussion on the ticket and decided to make a first step splitting the body